### PR TITLE
corrected inplace modication of house_indicies while been looped over and refactor masking logic in BERTDataset for improved performance and clarity

### PR DESCRIPTION
**Pull request by Mat-thias**
Editing the house_indicies to be read, this modifies the iterator, cause some houses to be skip, like for house 4 when working with the kettle continue is enough to skip the reading houses without the appliance self.house_indicies.remove(house_id)

I noticed when working with the ukdale dataset, with kettle, house 5 is skipped. This is due to house 4 which is just before house 5, does not have the appliance, so 4 is removed from the house_indicies
```python
self.house_indicies.remove(house_id)

# changes house_indicies from [1, 3, 4, 5] to [1, 3, 5]
```
so, house 5 never gets read

### CODE REVIEW

**Comment by Mat-thias** [comment]
# Refactor masking logic in BERTDataset for improved performance and clarity
The looping operation to mask the token in the fly is very expensive, I converted it to a vectorised form, with numpy which improved training speed.

snippet
```python
# setting 80% to -1
tokens[mask_idx_set_to_minus_1] = -np.ones_like(tokens[mask_idx_set_to_minus_1])
# setting 10% to guass noise
tokens[mask_idx_set_to_guass_noise] = np.random.normal(size=tokens[mask_idx_set_to_guass_noise].shape)
# the remaining 10% is left unchanged
```
@Yueeeeeeee

**Comment by Mat-thias** [comment]
## Problem
The previous KL-divergence implementation used torch.masked_select to apply masking before computing the loss. However, masked_select flattens the selected elements into a 1-D tensor, which causes samples from different temporal windows to be concatenated.

As a result:
- The KL divergence is computed across mixed windows
- Probability distributions are no longer window-aligned
- The loss unintentionally compares logits and labels belonging to different time steps / windows
- This violates the intended per-window distribution comparison described in the methodology

This behavior is subtle and silent, but leads to incorrect KL estimates.

## Fix
This PR refactors the KL-divergence computation to:
- Preserve the original window structure
- Apply masking without collapsing dimensions
- Compute KL divergence element-wise per window
- Correctly handle log-space targets using log_target=True

## Key changes:
- Replace masked_select with masked_fill
- Use reduction='none' to retain per-element losses
- Apply masking after KL computation
- Aggregate loss only over valid (unmasked) positions

@Yueeeeeeee
